released/usecases/devops/main.tf

```diff
@@ -0,0 +1,381 @@
+###############################################################################################
+# Setup subaccount domain and the CF org (to ensure uniqueness in BTP global account)
+###############################################################################################
+locals {
+  project_subaccount_domain = lower("${var.subaccount_name}")
+  project_subaccount_cf_org = substr(replace("${local.project_subaccount_domain}${var.org}", "-", ""),0,32)
+}
+
+resource "btp_directory" "proj" {
+  name        = var.directory
+  description = var.directory_desc
+  features    = ["DEFAULT"]
+  labels = var.directory_labels
+}
+ 
+###############################################################################################
+# Creation of subaccount
+###############################################################################################
+resource "btp_subaccount" "create_subaccount" {
+  count = var.subaccount_id == "" ? 1 : 0
+  name      = var.subaccount_name
+  subdomain = local.project_subaccount_domain
+  parent_id = btp_directory.proj.id
+  region    = lower(var.region)
+  labels    = var.subaccount_labels
+}
+
+data "btp_subaccount" "project" {
+  id = var.subaccount_id != "" ? var.subaccount_id : btp_subaccount.create_subaccount[0].id
+}
+
+data "btp_subaccount_environments" "all" {
+  subaccount_id = data.btp_subaccount.project.id
+}
+
+
+ 
+###############################################################################################
+# Assign custom IDP to subaccount
+###############################################################################################
+
+resource "btp_subaccount_trust_configuration" "simple" {
+  subaccount_id     = data.btp_subaccount.project.id
+  identity_provider = var.custom_idp
```

Comment:
Does this work if this is entered empty? Would a ternary expression be better here to check if something is provided and if not exclude the resource form execution